docs/source/implementations.rst

```diff
@@ -0,0 +1,92 @@
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. _implementations:
+
+===============
+Implementations
+===============
+
+Official Implementations
+========================
+
+The Apache Arrow project houses a collection of libraries for different
+programming languages. Use the links in the table below to access the
+documentation and source code for these libraries.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Language
+     - Docs
+     - Source
+   * - C++
+     - `C++ Docs <https://arrow.apache.org/docs/cpp>`_
+     - `C++ Source <https://github.com/apache/arrow/tree/main/cpp>`_
+   * - C GLib
+     - `C GLib Docs <https://arrow.apache.org/docs/c_glib>`_
+     - `C GLib Source <https://github.com/apache/arrow/tree/main/c_glib>`_
+   * - C#
+     - `C# Docs <https://github.com/apache/arrow/blob/main/csharp/README.md>`_
+     - `C# Source <https://github.com/apache/arrow/tree/main/csharp>`_
+   * - Go
+     - `Go Docs <https://godoc.org/github.com/apache/arrow/go/arrow>`_
+     - `Go Source <https://github.com/apache/arrow-go>`_
+   * - Java
+     - `Java Docs <https://arrow.apache.org/docs/java>`_
+     - `Java Source <https://github.com/apache/arrow-java>`_
+   * - JavaScript
+     - `JavaScript Docs <https://arrow.apache.org/docs/js>`_
+     - `JavaScript Source <https://github.com/apache/arrow-js>`_
+   * - Julia
+     - `Julia Docs <https://arrow.apache.org/julia/>`_
+     - `Julia Source <https://github.com/apache/arrow-julia>`_
+   * - MATLAB
+     - `MATLAB Docs <https://github.com/apache/arrow/blob/main/matlab/README.md>`_
+     - `MATLAB Source <https://github.com/apache/arrow/tree/main/matlab>`_
+   * - Python
+     - `Python Docs <https://arrow.apache.org/docs/python>`_
+     - `Python Source <https://github.com/apache/arrow/tree/main/python>`_
+   * - R
+     - `R Docs <https://arrow.apache.org/docs/r>`_
+     - `R Source <https://github.com/apache/arrow/tree/main/r>`_
+   * - Ruby
+     - `Ruby Docs <https://github.com/apache/arrow/blob/main/ruby/README.md>`_
+     - `Ruby Source <https://github.com/apache/arrow/tree/main/ruby>`_
+   * - Rust
+     - `Rust Docs <https://docs.rs/arrow/latest>`_
+     - `Rust Source <https://github.com/apache/arrow-rs>`_
+   * - Swift
+     - `Swift Docs <https://github.com/apache/arrow/blob/main/swift/Arrow/README.md>`_
+     - `Swift Source <https://github.com/apache/arrow/tree/main/swift>`_
+
+In addition to the libraries listed above, the Arrow project hosts the
+**nanoarrow** subproject which provides a set of lightweight libraries
+designed to help produce and consume Arrow data.
+
+.. list-table::
+   :header-rows: 0
+
+   * - nanoarrow
+     - `nanoarrow Docs <https://arrow.apache.org/nanoarrow>`_
+     - `nanoarrow Source <http://github.com/apache/arrow-nanoarrow>`_
+
```

Comment:
Could we add a section for the Cookbooks here as well? It would be helpful to make them accessible from multiple locations.

```suggestion
 C++ cookbook <https://arrow.apache.org/cookbook/cpp/>
 Java cookbook <https://arrow.apache.org/cookbook/java/>
 Python cookbook <https://arrow.apache.org/cookbook/py/>
 R cookbook <https://arrow.apache.org/cookbook/r/>
```

As a list-table or as a normal list, not sure.